docs/design/distributed-cache.md

```diff
@@ -0,0 +1,110 @@
+# Distributed cache design
+
+## Purpose
+
+The purpose of a distributed cache is to improve the performance and scalability of JSON-RPC Relay by storing frequently accessed data in memory, making it readily available for quick retrieval by all running instances. This will help functionality like Filter API to work properly as it's heavily dependent on cache.
+
+## Goals
+
+1. **Scalability**: Distributed caches can be used to distribute the load across multiple instances of an application. As the application scales, the cache can help maintain consistent performance by providing a shared data store that all instances can access.
+2. **Improved Performance**: By sharing cache accross multiple JSON-RPC Relay instances, correct data will always be available for methods to access upon requests.
+3. **Reduced Load on Mirror-Node**: By caching data and serving it directly from the cache, the load on the underlying data sources (such as Mirror-Node) is reduced. This can help prevent performance bottlenecks and overloading of these resources.
+
+## Architecture
+
+For distributed cache JSON-RPC Relay will be using Redis and in order to establish connection and manage, it needs to utilize `node-redis` package. All cache managment will be done in the respective cache client, either for the internal `lru-cache` or shared redis, which methods will be defined by an interface. Using dependency injection pattern with dependency inversion all services that require cache will receive only abstraction in the constructor. This will allow for flexible usage of the cache without additional coupling, depending on the service need.
```

Comment:
This part is not clear to me. It's hints to an either approach whereas I think an and approach is better,
Services should be able to use both a local and a shared cache at the same time. This would avoid needing to store some items in a shared node when it's really only useful locally. 
A relay instance should be able to continue to operate partially if access to a redis cache went down. So only shared resources should go to redis

Reply:
Updated the logic.